utilities/object_registry.cc

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#include "logging/logging.h"
+#include "rocksdb/env.h"
+#include "rocksdb/utilities/object_registry.h"
+
+namespace rocksdb {
+// Looks through the "type" factories for one that matches "name".
+// If found, returns the pointer to the Entry matching this name.
+// Otherwise, nullptr is returned
+const ObjectRegistry::Entry *ObjectRegistry::Library::FindEntry(const std::string &type,
+								const std::string &name) const {
+  auto entries = entries_.find(type);
+  if (entries != entries_.end()) {
+    for (const auto &entry : entries->second) {
+      if (entry->matches(name)) {
+        return entry.get();
+      }
+    }
+  }
+  return nullptr;
+}
+  
+void ObjectRegistry::Library::AddEntry(const std::string & type, std::unique_ptr<Entry> & entry) {
+  auto& entries = entries_[type];
```

Comment:
Before or after this call.   Before, yes; after no.  The documentation for map[x] says it will do an insert if the value does not exist (https://en.cppreference.com/w/cpp/container/unordered_map/operator_at).

Do you suggest a different construct?